src/main/java/com/kovbas/kafka_prioritized_queue/tasks/ConsumerTask.java

```diff
@@ -0,0 +1,186 @@
+package com.kovbas.kafka_prioritized_queue.tasks;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+
+@Component
+public class ConsumerTask implements ApplicationRunner {
+
+    @Value("#{${kafka_topics}}")
+    private List<String> topics;
+
+    @Value("#{${kafka_consumers_properties}}")
+    private List<Properties> consumersProperties;
+
+    private final List<ConsumerThread> consumerThreads = new ArrayList<>();
+
+
+    @Override
+    @Async
+    public void run(ApplicationArguments applicationArguments) throws Exception {
+
+        // Assert that number of topics is equals to number of consumers
+        // as we will create separate consumer for each topic
+        assert topics.size() == consumersProperties.size();
+
+        // Initialize and start threads
+        {
+
+            for (int i = 0; i < topics.size(); i++) {
+
+                ConsumerThread consumerThread = new ConsumerThread(
+                        topics.get(i),
+                        consumersProperties.get(i)
+                );
+
+                consumerThread.start();
+
+                // thread order represents priority
+                consumerThreads.add(consumerThread);
+            }
+        }
+
+
+
+        while (true) {
+
+            Thread.sleep(10); // only for testing purposes
+
+            for (ConsumerThread thread : consumerThreads) {
+
+                ConsumerRecords<String, String> records = thread.getConsumerRecords();
+
+                if (records != null && !records.isEmpty()) {
```

Comment:
The problem is that pull and process are done in separate threads. And those threads are not suspending when other is doing his job. Thus potential false-negatives will happen - process thread will go further to next topic (lower) because poll thread won't be able to poll data in between those checks made be processing thread. Does it make sense? Just imagine a case when you have millions of messages to process in high topic, so if processing is faster than poll - it will move down to lower topics all the time.